### ISSUE TITLE
Bufferoverflow when escaping characters

### ISSUE THREAD

**Issue by stijnvanraes** [pljson/pljson#14]
Hi 

I got a similar issue then issue #13, but in a different case.
When I execute this code:

```
declare 
      l_json  json := json();
      l_test  clob;
    begin
      for i in 1..10000 loop
        l_test := l_test || '<div class="table-wrapper">';
      end loop;
   
      l_json.put('clob', json_value(l_test))
      l_json.htp;
     
  end;
```

I get following error: 
```
ORA-06512: in &quot;PLJSON.JSON_PRINTER&quot;, regel 47
ORA-06512: in &quot;PLJSON.JSON_PRINTER&quot;, regel 201
ORA-06512: in &quot;PLJSON.JSON_PRINTER&quot;, regel 238
ORA-06512: in &quot;PLJSON.JSON_PRINTER&quot;, regel 256
ORA-06512: in &quot;PLJSON.JSON&quot;, regel 284
ORA-06502: PL&#x2F;SQL: numerieke fout of fout in waarde : character string buffer too small.
```

By tracing it down I found that in json_printer.ppMem on line 192 the amount of the buffer is set 32767. But when the string get's escaped in escapeString, it can easily exceed the buffer.
If I change the buffer amount to 819 - buffer size in escapeString(40)/32767 - then it works.
I don't know the impact on performance though, I can imagine you would need run some tests on that.

Greetings

**Comment by boriborm** [comment]
Hi, sorry for my english.
Function escapeString return VARCHAR2 type. But it bad idea. String for escaping can be, to example, 30000 chars, but after escaping it may be 50000 chars.
Need to replace type of escapeString to CLOB and change algoritms for it.

**Comment by jsumners** [comment]
`escapeString` is used by other `varchar2` based methods. So simply make it return a `clob` is not a solution. What really needs to happen is that wherever a call to `escapeString` is made, the data needs to be submitted such that it can't cause the buffer overflow. `escapeString` should also throw an exception to be handled by invoking methods.

**Comment by boriborm** [comment]
Hi, James

This need make other function escapeStringToClob and use it internal
json_printer package for pretty_print functions. When i will have free
time, i do it. I need this solution on my work.

Sorry for my english :)
15.01.2016 17:05 пользователь "James Sumners" <notifications@github.com>
написал:

**Comment by jsumners** [comment]
@boriborm we'll appreciate any solution you can supply.

**Comment by jsumners** [comment]
A fix was merged in #27. If this isn't working for you, you can re-open the issue.
